### PR TITLE
test/pod: add scripts for running e2e tests in a pod

### PR DESCRIPTION
**Pull request by hasbro17**
[skip ci]

Added some scripts and a REAME to show how to run the e2e tests inside a pod.

/cc @fanminshi @hongchaodeng PTAL and try out the README.

### CODE REVIEW

File: test/pod/build [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+if ! which docker > /dev/null; then
+	echo "docker needs to be installed"
+	exit 1
+fi
+
+: ${TEST_IMAGE:?"Need to set TEST_IMAGE, e.g. gcr.io/coreos-k8s-scale-testing/etcd-operator-tests"}
+
+echo "building container..."
+docker build --tag "${TEST_IMAGE}" -f test/pod/Dockerfile . #1>/dev/null
```
Comment (hongchaodeng) [inline test/pod/build]: Should ignore output

File: test/pod/build [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+# For gcr users, do "gcloud docker -a" to have access.
+echo "pushing container..."
+docker push "${TEST_IMAGE}" #1>/dev/null
```
Comment (hongchaodeng) [inline test/pod/build]: remove comment

File: test/pod/run-test-pod [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+# This script runs the test-pod with the specified TEST_IMAGE and necessary envs for e2e tests
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+: ${TEST_IMAGE:?"Need to set TEST_IMAGE, e.g. gcr.io/coreos-k8s-scale-testing/etcd-operator-tests"}
+: ${TEST_NAMESPACE:?"Need to set TEST_NAMESPACE"}
+
+# TODO: Add upgrade pass
```
Comment (hongchaodeng) [inline test/pod/run-test-pod]: remove this todo.
Create an issue

File: test/pod/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# TODO: Use golang:1.9-alpine (and add necessary pkgs for go test) to get a smaller image size.
```
Comment (hongchaodeng) [inline test/pod/Dockerfile]: What does this TODO mean?

**Comment by hasbro17** [comment]
fixed

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**